src/validation.cpp

```diff
@@ -4458,12 +4458,13 @@ static bool ContextualCheckBlockHeader(const CBlockHeader& block, BlockValidatio
     if (IsThisSoftwareExpired(block.nTime)) {
         // Wait an extra day before we start rejecting blocks
         CBlockIndex const *blockindex_old = pindexPrev;
-        for (int i = 0; i < 144; ++i) {
+        for (int i = 0; i < std::min(144, pindexPrev->nHeight); ++i) {
```

Comment:
I suppose it should be
```suggestion
        for (int i = 0; i < std::min(144, pindexPrev->nHeight - 1); ++i) {
```
?

But to avoid re-calculating it every iteration, better to offset the starting index.